lib/vertica/query.rb

```diff
@@ -83,7 +84,8 @@ def handle_data_row(message)
   def handle_command_complete(message)
     if buffer_rows?
       @result = Vertica::Result.new(row_description: @row_description, rows: @buffer, tag: message.tag)
-      @row_description, @buffer = nil, nil
+      @row_description = nil
+      @buffer.clear if @enable_buffering
```

Comment:
I want to get an error if a former statement gets errored, and currently it is working so. I added tests to show the spec.